plugins/inputs/slab/slab.go

```diff
@@ -0,0 +1,32 @@
+package slab
+
+import (
+	"os"
+	"path"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type SlabStats struct {
+	Log telegraf.Logger `toml:"-"`
+
+	statFile string
+	UseSudo  bool `toml:"use_sudo"`
```

Comment:
Is /proc/slabinfo ever readable by non-root user? I mean does it ever make sense for UseSudo to be false? We don't want to encourage users to run telegraf as root. Maybe we should only ever use sudo and not make it possible to run without. What do you think?

Reply:
I thought about that same thing; however, I think it is better that our plugins attempt to have some consistency across the board: plugins that require sudo explicitly require the option.

Reply:
For the other plugins that have a use_sudo option, there is a non-sudo way to provide the extra access without running telegraf as root. Sometimes you add the telegraf user to a group, sometimes you grant a capability to the telegraf binary with setcap. The plugin's readme should describe how to set up non-root access.

I just read through a few of the proc related manpages and found this in 'man 5 slabinfo' (https://man7.org/linux/man-pages/man5/slabinfo.5.html):
> Only root can read and (if the kernel was configured with CONFIG_SLAB) write the /proc/slabinfo file.

Other proc man pages describe ways to access other sections of /proc as non-root, but it looks like slabinfo in particular just doesn't have a way besides running as root (or running cat as root through sudo).

If it's true that there just isn't another way to access /proc/slabinfo without being root, then this plugin is a different situation than the other plugins that have a use_sudo option. It's not a matter of being consistent with them because they have a safe non-root way, and this doesn't. If use_sudo is false (the current default) the plugin will only work if telegraf is run as root. That's something we strongly discourage.

Unless we can find and document a safe way without using sudo we shouldn't add a use_sudo setting, and instead change it always to use sudo so we don't encourage people to run telegraf as root.

Reply:
> It's much easier to add a setting a setting later than deprecate and remove it.

I get that, but I still disagree. You are creating a scenario where a plugin does have a behavior different than other plugins. Yes, it will always require root, but these types of little exceptions are what lead to confusion and continue to create a maintenance burden.

Reply:
Thanks for your review.
A quick check (`git grep -i Command.*sudo`) shows that 8 of existing input plugins were using sudo, and [only one of them](https://ghe.corp.yahoo.co.jp/ynw/telegraf/blob/6dc2b99549b3fc4684295ebcd2975cee96a1fa28/plugins/inputs/dmcache/dmcache_linux.go#L179) was forcing the use of sudo without `use_sudo` flag.
But, I have no strong opinion on either approach. If necessary, I will fix the patch.

Reply:
> [only one of them](https://ghe.corp.yahoo.co.jp/ynw/telegraf/blob/6dc2b99549b3fc4684295ebcd2975cee96a1fa28/plugins/inputs/dmcache/dmcache_linux.go#L179) was forcing the use of sudo without use_sudo flag.

*sigh* it seems we already do have his pattern with dmcache. Go ahead and make the change to always run with sudo.

Reply:
It doesn't matter what dmcache does, consistency is not more important than preventing users from shooting themselves in the foot. A line or two in the documentation and sample config will clear up any confusion from not having a setting.

Reply:
Yes, I have changed it to always use sudo. I appreciate both of your comments.